parser/src/cfg/chord.rs

```diff
@@ -10,105 +15,328 @@ pub(crate) fn parse_defchordv2(
+                    .next()
+                    .map(|kc| kc.to_string().to_lowercase())
+                    .zip(
+                        idx.try_into()
+                            .ok()
+                            .and_then(OsCode::from_u16)
+                            .map(|osc| osc.to_string().to_lowercase()),
+                    )
             })
-            .ok_or_else(|| {
-                anyhow_expr!(
-                    disabled_layers,
-                    "Disabled layers must be a list of layer names"
-                )
-            })??;
-        let chord = ChordV2 {
-            action,
-            participating_keys: s.a.sref_vec(participants.clone()),
-            pending_duration: timeout,
-            disabled_layers: s.a.sref_vec(disabled_layers),
+            .collect::<Vec<_>>()
+            .into_iter()
+            .chain(vec![(" ".to_string(), "spc".to_string())])
+            .collect::<HashMap<_, _>>();
+        ChordTranslation {
+            file_name,
+            target_map,
+            postprocess_map,
+            timeout,
             release_behaviour,
-        };
-        let chord = s.a.sref(chord);
-        for pkey in participants.iter().copied() {
-            log::trace!("chord for key:{pkey:?} > {chord:?}");
-            chords_container
-                .mapping
-                .entry(pkey)
-                .or_insert(ChordsForKey { chords: vec![] })
-                .chords
-                .push(chord);
+            disabled_layers,
         }
     }
-    let rem = chunks.remainder();
-    if !rem.is_empty() {
-        bail_expr!(
-            rem.last().unwrap(),
-            "Incomplete chord entry. Each chord entry must have 5 items:\n\
-        particpating-keys, action, timeout, release-type, disabled-layers"
+
+    fn post_process(&self, converted: &str) -> String {
+        self.postprocess_map
+            .get(converted)
+            .map(|c| c.to_string())
+            .unwrap_or_else(|| {
+                if converted.chars().all(|c| c.is_uppercase()) {
+                    format!("S-{}", converted.to_lowercase())
+                } else {
+                    converted.to_string()
+                }
+            })
+    }
+
+    fn participant_keys(&self, keys: &str) -> Vec<String> {
+        keys.chars()
+            .map(|key| {
+                self.target_map
+                    .get(key.to_string().to_lowercase().as_str())
+                    .map(|c| self.postprocess_map.get(c).unwrap_or(c).to_string())
+                    .unwrap_or_else(|| key.to_string())
+            })
+            .collect::<Vec<String>>()
+    }
+
+    fn action(&self, action: &str) -> Vec<String> {
+        let mut action_strings = action
+            .chars()
+            .map(|c| self.post_process(&c.to_string()))
+            .collect_vec();
+        // Wait 50ms for one-shot Shift to release
+        // TODO: This would be better handled by a (multi (release-key lsft)(release-key rsft))
```

Comment:
This would rely on a `tap-vkey`. For a "worse-is-better" solution, using this functionality could mandate a virtual key with a specific name that does the specific function 😅. I haven't thought too hard about it.

Reply:
This relates to a general concern I've been thinking about:
Mapping all the functionality from a ZipChord chord definition requires more than just entries in the `defchordsv2-experimental` section. E.g. the suffix stuff is currently handled via sequences which are top-level in the config. The `tap-vkey` is another indicator that the chording might have to live in the top-level config, so that it can instantiate everything required for a seamless experience, and require the user to add magic stuff elsewhere in his config.

I'd be very thankful for your ideas and opinion here @jtroo.

Reply:
To address this point specifically, I would be fine with to have the documentation+error message mandate an item like below to use `include`:
```
(defvirtualkeys release-shift (multi (release-key lsft) (release-key rsft)))
```
The code could even validate that the action matches exactly too. The information required to do these validations is found in `ParserState.virtual_keys`. The generated `macro` that gets created would then use a `on-press tap-vkey release-shift`.

---

At a higher level, my overall impression is that this PR is a great step in improving the convenience of using today's chords. I would merge it when it's ready, if we want to keep it as-is. However, to get even closer to ZipChord-like or Plover-like functionality, it would IMO be best done with a dedicated feature. As discovered in our experimentation, stitching together Kanata's existing features - chordsv2+sequences - to emulate ZipChord doesn't work perfectly.

My first thought would be to port https://github.com/psoukie/zipchord/blob/main/source/io.ahk as highlighted by psoukie into the output feed of kanata at https://github.com/jtroo/kanata/blob/main/src/kanata/output_logic.rs . At least from my initial impression ZipChord seems easier to port than Plover, so seems like a good starting point. On the point of _where_ the implementation would be, my feeling is that adapting/extending the ZMK-inspired chording, which is done at the Keyberon state machine level, is probably not preferable.